src/AWSS3.jl

```diff
@@ -410,6 +410,44 @@ end
 
 s3_delete(a...; b...) = s3_delete(global_aws_config(), a...; b...)
 
+"""
+    s3_delete_all_versions([::AbstractAWSConfig], bucket, path; kwargs...)
+
+Deletes all versions of object `path` from `bucket`, forwarding all `kwargs` to the
+internal per-version [`s3_delete`](@ref) call.
+
+To only delete one specific version, see [`s3_delete`](@ref); to delete all versions
+EXCEPT the latest version, see [`s3_purge_versions`](@ref).
+
+# API Calls
+
+- [`DeleteObject`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObject.html)
+- [`ListObjectVersions`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectVersions.html)
+
+# Permissions
+
+- [`s3:DeleteObject`](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazons3.html#amazons3-DeleteObject)
+- [`s3:ListBucketVersions`](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazons3.html#amazons3-ListBucketVersions)
+"""
+function s3_delete_all_versions(aws::AbstractAWSConfig, bucket, path; kwargs...)
```

Comment:
I've renamed it to `s3_nuke_object_versions` to prevent path v prefix confusion around the difference in behavior from `s3_purge_versions` (discussed in previous comments).

...could also switch to `s3_nuke_object`....

Reply:
(okay, there's now a naming mismatch between this function and its aws resources---i'll rename those once the updated name is deemed sufficient)

Reply:
rename complete! (resources also renamed, so may need another @omus push...)